wcfsetup/install/files/proxy_sourcemap.php

```diff
@@ -0,0 +1,230 @@
+ */
+function verifySignature(string $map, string $signature): bool
+{
+    $publicKey = "-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA45HcF7xOmGe0TSwVwCHE
+hFYYxD5SgxchDwQSbX4Wa0XrvhJpo7yKy2QWlfc3CfXZDTLIqPMqAvJEd7xTP9Ny
+5OF6cR2NDPjR/ilGN26txhTc2BNzOSXBMZyIwhKHYZZ2JThMT2MRmsVjFeJjbPrX
+d8ttf4rEox+ARY/Vaoq+1nx8ZB2B/SwiOS18ESVKRbtKcGtJvenMYxQRx+n7iqy3
+ALh9/pSWX5iucJvmW4bBx+RZAn9eTrAjWf5y8Yadc7sMOFWNg2zCgIqqPpsA5ccq
+Di1RWwi7vdSudukBovTVfhQ1CkTQ4/r6YxQTYJE2JvyRCMeTEsUHa2Kmwndj/nX3
+xQIDAQAB
+-----END PUBLIC KEY-----";
+
+    return \openssl_verify(
+        $map,
+        $signature,
+        $publicKey,
+        \OPENSSL_ALGO_SHA256
+    ) === 1;
+}
+
+/**
+ * Extracts the map name from the given $query.
+ *
+ * Returns a string if a map name could be extracted. Returns a class implementing
+ * ResponseInterface if the $query is invalid.
+ */
+function getMapFromQuery(string $query)
+{
+    try {
+        // Step 1: Extract the signature and map name.
+
+        $parts = \explode('/', $query, 2);
+        if (\count($parts) !== 2) {
+            throw new \UnexpectedValueException('Expected exactly 2 parts within the query string.');
+        }
+
+        [$signature, $map] = $parts;
+
+        $signature = Base64UrlSafe::decode($signature, false);
+
+        // Step 2: Verify the signature.
+        if (!verifySignature($map, $signature)) {
+            throw new \UnexpectedValueException('Failed to verify the signature.');
+        }
+
+        // Step 3: Perform a safety check on the map name.
+        if (!\preg_match('/^[a-zA-Z0-9]+(\\.[a-zA-Z0-9]+)*\\/[a-f0-9]{64}$/', $map)) {
+            // This should be unreachable in real world, because an invalid map is not
+            // going to be correctly signed.
+            return new Response(
+                400,
+                [
+                    'content-type' => 'text/plain',
+                    'cache-control' => [
+                        'public',
+                    ],
+                ],
+                'Invalid map.'
+            );
+        }
+
+        // Step 4: Return the extracted map name.
+        return $map;
+    } catch (\UnexpectedValueException | \RangeException $e) {
+        return new Response(
+            400,
+            [
+                'content-type' => 'text/plain',
+                'cache-control' => [
+                    'public',
+                ],
+            ],
+            'Invalid signature.'
+        );
+    }
+}
+
+/**
+ * Processes the given request and returns a response to send to the browser.
+ */
+function handle(ServerRequestInterface $request): ResponseInterface
+{
+    $mapOrErrorResponse = getMapFromQuery($request->getUri()->getQuery());
+
+    if ($mapOrErrorResponse instanceof ResponseInterface) {
+        return $mapOrErrorResponse;
+    }
+
+    $map = $mapOrErrorResponse;
+
+    $cacheFilename = \sprintf(
+        '%s/tmp/sourcemap_%s.map',
```

Comment:
Why did you opted to use `tmp/` and do not use `FileUtil::getTempFolder()` (which would probably make this more error resilient)? :)

Reply:
Because this script very intentionally does not use anything from within WoltLab Suite and `tmp/` within WoltLab Suite Core's directory effectively always exists.